src/api/transactions.coffee

```diff
@@ -317,3 +393,28 @@ exports.removeTransaction = (transactionId) ->
     logger.info "User #{this.authenticated.email} removed transaction with id #{transactionId}"
   catch e
     utils.logAndSetResponse this, 'internal server error', "Could not remove transaction via the API: #{e}", 'error'
+
+
+
+###
+# construct transaction filtering dropdown options
+###
+exports.getTransactionsFilterOptions = ->
+
+  # Must be admin
+  if not authorisation.inGroup 'admin', this.authenticated
+    utils.logAndSetResponse this, 'forbidden', "User #{this.authenticated.email} is not an admin, API access to getTransactionsFilterOptions denied.", 'info'
+    return
+
+  try
+    # execute the query
+    channels = yield Channel.find().exec()
+    clients = yield Client.find().exec()
+
+    responseObject =
+      channels: channels
+      clients: clients
```

Comment:
I originally had it as two API calls but though it might be better to make one call and return all the filters needed. Reason for this is that it would be easier to add more filter options to this return if needed in the future